src/types/clients.ts

```diff
@@ -13,30 +13,34 @@ export type GenLayerMethod =
   | {method: "gen_getContractSchemaForCode"; params: [contractCode: string]}
   | {method: "eth_getTransactionCount"; params: [address: string]};
 
-export type GenLayerClient<
-  TTransport extends Transport,
-  TSimulatorChain extends SimulatorChain,
-  TAccount extends Account,
-> = Client<TTransport, TSimulatorChain, TAccount> & {
-  request: Client<TTransport, TSimulatorChain, TAccount>["request"] & {
-    <TMethod extends GenLayerMethod>(
-      args: Extract<GenLayerMethod, {method: TMethod["method"]}>,
-    ): Promise<unknown>;
+export type GenLayerClient<TSimulatorChain extends SimulatorChain> = Omit<
+  Client<Transport, TSimulatorChain>,
+  "transport" | "getTransaction" | "readContract"
+> &
+  Omit<PublicActions<Transport, TSimulatorChain>, "readContract" | "getTransaction"> & {
```

Comment:
Please add a comment to this type to explain why it is so complex. Something like 
"The `Omit`s are done to override some methods with new types"

Reply:
Done! 
This means:
Take all the properties from PublicActions<Transport, TSimulatorChain>
Remove the readContract and getTransaction properties
The resulting type will have everything from PublicActions EXCEPT those two properties
When using the & (intersection) operator to combine types, if there are properties with the same name but different types, TypeScript will try to create an intersection of those property types. If the types are incompatible, you'll get a never type or other conflicts.
By using Omit, you're explicitly:
Removing the original properties first
Then adding your new versions of those properties with different types/signatures
This is a common pattern when you want to override specific properties while keeping the rest of the type's structure intact. Without the Omit, you'd have a type conflict between the original readContract/getTransaction and your new implementations.